### PR TITLE
Clean up experimental_lambdify

### LINKED ISSUE #19961: The adaptive plotting algorithm appears not consistent
The following code

    from matplotlib.pyplot import rc_context
    from sympy import pi, plot, sin, sqrt, symbols
    a, t, w0 = symbols('a, t, omega_0')
    q = -26*sin(w0*t/2) + 13*sqrt(39)*sin(sqrt(39)*w0*t/13)/3
    display(q)
    with rc_context({'figure.figsize': (10, 2)}):
        nondimtime = {t:a/w0}
        plot(q.subs(nondimtime)/26, (a, 0, 60*pi), xlabel=r'$a=\omega_0t$')
        plot(q.subs(nondimtime),    (a, 0, 60*pi), xlabel=r'$a=\omega_0t$')
        nondimtime = {t:pi*a/w0}
        plot(q.subs(nondimtime),    (a, 0, 60*1.), xlabel=r'$a=\frac{\omega_0t}{\pi}$')

plots three times the same function but with different assumptions wrt scaling and when I run it in a Jupyter notebook i get

image

1st plot: I normalize my function to `-1, 1` and I get a really smooth plot.
2nd plot: I plot the function with its own amplitude and I can see a slight jagginess, e.g, the maximum before `a=125`.
3rd plot: I normalize the abscissae with respect with π and the situation degenerates with respect to jagginess.

I expected a more consistent behavior from the adaptive algorithm.

### PR DESCRIPTION
**Pull request by sylee957**
#### Brief description of what is fixed or changed

This is the first stage to clean up `experimental_lambdify` which had became a huge unmaintainable obstacle to improve the plotting module.
I've found that the errors like `no ordering relation is defined for complex numbers` only happens when comparison with real numbers is used inside piecewise, which can be resolved by not wrapping the inputs with `complex`.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by asmeurer** [comment]
Improvements here are good, but I think our long term aim should be to remove experimental_lambdify and replace it with normal lambdify.

**Comment by sylee957** [comment]
I agree that experimental_lambdify should be deprecated, but unfortunately, there are too many obstacles to replace it with lambdify for now.

**Comment by asmeurer** [comment]
We should create a tracking issue for it. I know I've mentioned it a few times, but I don't see an actual issue for it. The problem with experimental_lambdify is that it is an ad hoc reimplementation of lambdify. lambdify has since had a lot of improvements to the way it generates a lot of corner cases, which experimental_lambdify doesn't support.

File: sympy/plotting/experimental_lambdify.py [hunk 1]
```diff
@@ -107,68 +107,41 @@ class vectorized_lambdify(object):
...
-        finally:
-            np.seterr(**np_old_err)
-
-        return results
+                np.abs(results.imag) > 1e-7 * np.abs(results),
+                results.real, copy=False)
+            return results
+        except ValueError as e:
+            if self.failure:
+                raise e
```
Comment (eric-wieser) [inline sympy/plotting/experimental_lambdify.py:136]:
```suggestion
                raise
```
results in simpler tracebacks

Comment (asmeurer) [reply sympy/plotting/experimental_lambdify.py:136]: I've never noticed that difference in the two forms. You'll also have to get rid of the `as e` above to make flake8 happy.

**Comment by oscarbenjamin** [comment]
I don't know this code so well but these changes look good to me

**Comment by oscarbenjamin** [comment]
@sylee957 are you intending to do further work on this?

+1 to merge from me

**Pull request merged**